internal/database/db.go

```diff
@@ -0,0 +1,94 @@
+package database
+
+import (
+	"fmt"
+	"github.com/cyneruxyz/address-book/gen/proto"
+	"github.com/cyneruxyz/address-book/internal/database/model"
+	"github.com/spf13/viper"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+var fieldModel = &model.AddressField{}
+
+type Database struct {
+	orm *gorm.DB
+}
+
+func New(conf *viper.Viper) (*Database, error) {
+	dsn := fmt.Sprintf(
+		"user=%s password=%s dbname=%s port=%s sslmode=%s TimeZone=%s",
+		conf.GetString("DB_USER"),
+		conf.GetString("DB_PASSWORD"),
+		conf.GetString("DB_NAME"),
+		conf.GetString("DB_PORT"),
+		conf.GetString("DB_SSLMODE"),
+		conf.GetString("DB_TIMEZONE"),
+	)
+
+	db, err := gorm.Open(postgres.New(
+		postgres.Config{
+			DSN:                  dsn,
+			PreferSimpleProtocol: true, // disables implicit prepared statement usage
+		}),
+		&gorm.Config{},
+	)
+
+	return &Database{db}, err
+}
+
+func (db *Database) Migrate() error {
+	return db.orm.AutoMigrate(fieldModel)
+}
+
+func (db *Database) CreateItem(field *proto.AddressField) error {
+	return db.orm.Create(fieldModel.Prepare(field)).Error
+}
+
+func (db *Database) ReadItem(param string) (fields []*proto.AddressField, err error) {
+	var items []model.AddressField
+
+	err = db.orm.Model(fieldModel).
```

Comment:
Wildcards should be considered here as well. Please use LIKE operator for queries here.